tests/phpunit/tests/dependencies/scripts.php

```diff
@@ -63,6 +66,699 @@ public function test_wp_enqueue_script() {
+
+		// Test defaults.
+		$expected_args = array(
+			'in_footer' => true,
+			'strategy'  => 'blocking',
+		);
+		wp_register_script( 'defaults-strategy', '/defaults.js', array(), null, array( 'in_footer' => true ) );
+		$this->assertSame( $expected_args, $wp_scripts->get_data( 'defaults-strategy', 'script_args' ) );
+
+		$expected_args = array(
+			'in_footer' => false,
+			'strategy'  => 'async',
+		);
+		wp_register_script( 'defaults-in-footer', '/defaults.js', array(), null, array( 'strategy' => 'async' ) );
+		$this->assertSame( $expected_args, $wp_scripts->get_data( 'defaults-in-footer', 'script_args' ) );
+
+		// scripts_args not set of args parameter is empty.
+		wp_register_script( 'empty-args-array', '/defaults.js', array(), null, array() );
+		$this->assertSame( false, $wp_scripts->get_data( 'defaults', 'script_args' ) );
+
+		wp_register_script( 'no-args', '/defaults.js', array(), null );
+		$this->assertSame( false, $wp_scripts->get_data( 'defaults-no-args', 'script_args' ) );
+
+		// Test backward compatibility.
+		$expected_args = array(
+			'in_footer' => true,
+			'strategy'  => 'blocking',
+		);
+		wp_enqueue_script( 'footer-old', '/footer-async.js', array(), null, true );
+		$this->assertSame( $expected_args, $wp_scripts->get_data( 'footer-old', 'script_args' ) );
+	}
+
+	/**
+	 * Test script strategy doing it wrong.
+	 *
+	 * For an invalid strategy defined during script registration, default to a blocking strategy.
+	 *
+	 * @ticket 12009
+	 */
+	public function test_script_strategy_doing_it_wrong() {
+		$this->setExpectedIncorrectUsage( 'WP_Scripts::get_intended_strategy' );
+
+		wp_register_script( 'invalid-strategy', '/defaults.js', array(), null, array( 'strategy' => 'random-strategy' ) );
+		wp_enqueue_script( 'invalid-strategy' );
+
+		$output = get_echo( 'wp_print_scripts' );
+
+		$expected = "<script type='text/javascript' src='/defaults.js' id='invalid-strategy-js'></script>\n";
+
+		$this->assertSame( $expected, $output );
+	}
+
+	/**
+	 * Test script concatenation with deferred main script.
+	 *
+	 * @ticket 12009
+	 */
+	public function test_concatenate_with_defer_strategy() {
+		global $wp_scripts, $concatenate_scripts;
+
+		$old_value           = $concatenate_scripts;
+		$concatenate_scripts = true;
+
+		$wp_scripts->do_concat    = true;
+		$wp_scripts->default_dirs = array( $this->default_scripts_dir );
+
+		wp_register_script( 'one-concat-dep', $this->default_scripts_dir . 'script.js' );
+		wp_register_script( 'two-concat-dep', $this->default_scripts_dir . 'script.js' );
+		wp_register_script( 'three-concat-dep', $this->default_scripts_dir . 'script.js' );
+		wp_enqueue_script( 'main-defer-script', '/main-script.js', array( 'one-concat-dep', 'two-concat-dep', 'three-concat-dep' ), null, array( 'strategy' => 'defer' ) );
+
+		wp_print_scripts();
+		$print_scripts = get_echo( '_print_scripts' );
+
+		// reset global before asserting.
```

Comment:
```suggestion
		// Reset global before asserting.
```
🔢 Applies elsewhere in the tests.

Reply:
Done in b708151c1f44f0bd32a01fc6a862ff6b9d067c82